torch/_inductor/codegen/cpp.py

```diff
@@ -3858,25 +4036,127 @@ def wrap_fn(fn):
     def codegen_loops(self, code, worksharing):
         self.codegen_loops_impl(self.loop_nest, code, worksharing)
 
+    def update_stores_with_parallel_reduction(self):
+        for kernel in self.kernels:
+            kernel.update_stores_with_parallel_reduction()
+
+    def gen_body(self, code: Optional[BracesBuffer] = None):
+        assert code is not None
+        if_prefix = "C10_LIKELY"
+        for kernel in self.kernels:
+            with contextlib.ExitStack() as stack:
+                if kernel.codegen_conditions(code, if_prefix):
+                    if_prefix = "C10_UNLIKELY"
+                    stack.enter_context(code.indent())
+                    code.splice(kernel.gen_body())
+
+    def aggregate_reduction_buffers(
+        self, inner_loop_reduction_outer_not: bool, outer_loop: Optional["LoopLevel"]
+    ):
+        # CppKernel/CppVecKernel/CppTile2dKernel have reduction buffers themselves.
+        # Here, we decide how to aggregate them together and place new reduction buffers
+        # under CppKernelProxy.
+        def set_buffer_from_main_loop_kernel(attr: str):
+            # the attr: str here is the name of the reduction buffers
+            assert attr in (
+                "reduction_prefix",
+                "reduction_suffix",
+                "parallel_reduction_prefix",
+                "parallel_reduction_suffix",
+                "local_reduction_init",
+                "local_reduction_stores",
+                "non_parallel_reduction_prefix",
+            )
```

Comment:
Instead of passing `str` attr and adding assertion for checking, I feel it is more type-safe to just invoke the attr with dot operator, e.g.
`self.reduction_prefix.splice(main_kernel.reduction_prefix)`

Just calling splice like this at the caller side without the need of such a function would be simpler.

Reply:
Thanks for advice, updated.